xbmc/pvr/channels/PVRChannel.cpp

```diff
@@ -111,7 +111,7 @@ bool CPVRChannel::Delete()
   const std::shared_ptr<CPVREpg> epg = GetEPG();
   if (epg)
   {
-    CServiceBroker::GetPVRManager().EpgContainer().DeleteEpg(epg);
+    CServiceBroker::GetPVRManager().EpgContainer().DeleteEpg(epg, true);
```

Comment:
The only place calling this with commit=true after this PR would be CPVREpgContainer::UpdateEPG and even there it would make sense to call it with commit=false.